frontend/src/container/LogControls/index.tsx

```diff
@@ -72,12 +75,61 @@ function LogControls(): JSX.Element | null {
 			type: GET_PREVIOUS_LOG_LINES,
 		});
 	};
+
 	const handleNavigateNext = (): void => {
 		dispatch({
 			type: GET_NEXT_LOG_LINES,
 		});
 	};
 
+	const flattenObject = useCallback(
+		(obj: ILog) =>
+			Object.entries(obj).reduce((flattenedObj, [key, value]) => {
+				// eslint-disable-next-line @typescript-eslint/no-explicit-any
+				const updatedObj: any = { ...flattenedObj };
+				if (typeof value === 'object' && value !== null) {
+					updatedObj[key] = JSON.stringify(value);
+				} else {
+					updatedObj[key] = value;
+				}
+				return updatedObj;
+			}, {}),
+		[],
+	);
+
+	const downloadExcelFile = useCallback((): void => {
+		console.log('Write a code here for excel download');
+	}, []);
+
+	const downloadCsvFile = useCallback((): void => {
+		const flattenLogData = logs.map((entry) => flattenObject(entry));
+		const csv = Papa.unparse(flattenLogData);
+		const csvBlob = new Blob([csv], { type: 'text/csv;charset=utf-8;' });
+		const csvUrl = URL.createObjectURL(csvBlob);
+		const downloadLink = document.createElement('a');
```

Comment:
this process can block the thread add loader and see if can use the already created element or remove the node from DOM after the completion